### PR TITLE
TBDGen: Add resilient enum case emission support

### PR DESCRIPTION
**Pull request by aschwaighofer**
rdar://38707747

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test and merge

File: lib/TBDGen/TBDGen.cpp [hunk 1]
```diff
@@ -299,6 +299,17 @@ void TBDGenVisitor::visitProtocolDecl(ProtocolDecl *PD) {
 #endif
 }
 
+void TBDGenVisitor::visitEnumDecl(EnumDecl *ED) {
+  if (!ED->isResilient(SwiftModule, ResilienceExpansion::Minimal))
```
Comment (jrose-apple) [inline lib/TBDGen/TBDGen.cpp]: This is just `isResilient()`. The module you're supposed to pass is the client module, not the library module, but ResilienceExpansion::Minimal makes that uninteresting anyway.

Comment (aschwaighofer) [reply lib/TBDGen/TBDGen.cpp]: Like so: #15799

**Comment by huonw** [comment]
Thanks!

**Pull request merged**